client/hwlib/src/collectors/hardware_info.rs

```diff
@@ -0,0 +1,338 @@
+/* Copyright 2024 Canonical Ltd.
+ * All rights reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Written by:
+ *        Nadzeya Hutsko <nadzeya.hutsko@canonical.com>
+ */
+
+use smbioslib;
+use time::macros::format_description;
+use time::Date;
+
+use crate::constants::PROC_DEVICE_TREE_DIRPATH;
+use crate::models::devices;
+
+pub fn load_smbios_data(
+    entry_filepath: Option<&'static str>,
+    table_filepath: Option<&'static str>,
+) -> Result<Option<smbioslib::SMBiosData>, Box<dyn std::error::Error>> {
```

Comment:
You only have the two return cases from the match here and no paths that return an `Error` so I'd suggest having this function just return `Option<smbioslib::SMBiosData>`.

Reply:
Fixed in fed34d